fix timeout problems / refactor OdooClient.cs

bartimaeusnek opened the pull request.

bartimaeusnek committed:
fix timeout problems / refactor OdooClient.cs

zivillian commented on PortaCapena.OdooJsonRpcClient/OdooClient.cs:
```diff
@@ -20,108 +18,81 @@ namespace PortaCapena.OdooJsonRpcClient
         public async Task<OdooResult<T[]>> GetAsync<T>(OdooQuery query = null, OdooContext context = null, CancellationToken cancellationToken = default) where T : IOdooModel, new()
         {
-            return await ExecuteWitrAccesDenideRetryAsync(userUid => GetAsync<T>(userUid, query, SelectContext(context, Config.Context), cancellationToken));
+            return await ExecuteWitrAccesDenideRetryAsync(userUid => GetAsync<T>(userUid, query, SelectContext(context, Config.Context), cancellationToken), cancellationToken);
```
die Tippfehler könnten wir auch gleich korrigieren

bartimaeusnek committed:
minor cleanup:
- Fix typos
- Remove unneeded async from wrapper methods
- Simplified GetTableName & SelectContext
